File: src/test/java/boat/DownloadMonitorIntegrationTest.java
package boat;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;

import boat.torrent.TorrentSearchEngineService;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertTrue;

@SpringBootTest
class DownloadMonitorIntegrationTest {

    @Autowired
    DownloadMonitor downloadMonitor;

    @Autowired
    TorrentSearchEngineService torrentSearchEngineService;

    @Test
    void refreshTorrentSearchEngines() {
        // Given
        // When
        downloadMonitor.refreshTorrentSearchEngines();
        // Then
        assertTrue(torrentSearchEngineService.getActiveSearchEngines().size() > 0);

    }

}

File: src/test/java/boat/DownloadMonitorIntegrationTest.kt
package boat

import boat.torrent.TorrentSearchEngineService
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest

@SpringBootTest
internal class DownloadMonitorIntegrationTest {
    @Autowired
    var downloadMonitor: DownloadMonitor? = null

    @Autowired
    var torrentSearchEngineService: TorrentSearchEngineService? = null
    @Test
    fun refreshTorrentSearchEngines() {
        // Given
        // When
        downloadMonitor!!.refreshTorrentSearchEngines()
        // Then
        Assertions.assertTrue(torrentSearchEngineService!!.getActiveSearchEngines().size > 0)
    }
}